E2E: Remove unnecessary presence checks from clickMySites helper

WunderBart opened the pull request:
#### Changes proposed in this Pull Request

Remove unnecessary element presence checks from clickMySites helper. Waiting for those elements causes an additional delay of ~20 seconds on each call (tested locally). That helper is used 17 times in our suites which means this tweak should save us a lot of time :)

_update_: Re-run [the build](https://teamcity.a8c.com/viewType.html?buildTypeId=calypso_RunCalypsoE2eDesktopTests&branch_calypso_WebApp=fix%2Fe2e-click-my-sites&tab=buildTypeStatusDiv) a few times - looks like we're consistently 🟢 and ~10-25% faster than [trunk](https://teamcity.a8c.com/viewType.html?buildTypeId=calypso_RunCalypsoE2eDesktopTests&branch_calypso_WebApp=%3Cdefault%3E&tab=buildTypeStatusDiv)!  🎉

#### Testing instructions

All specs should pass.

WunderBart committed:
Remove unnecessary presence checks from clickMySites helper
Waiting for those elements seems to be unnecessary and causes additional delay of 20 seconds on each call (tested locally). That helper is used 17 times in our suites which means this tweak should save us a lot of time :)

noahtallen commented on test/e2e/lib/components/nav-bar-component.js:
```diff
@@ -41,14 +41,6 @@ export default class NavBarComponent extends AsyncBaseContainer {
-			this.driver,
-			by.css( '.sidebar__menu-wrapper' )
-		);
-		return await driverHelper.isEventuallyPresentAndDisplayed(
```
Out of curiosity, why are they unnecessary? It seems like we are just making sure that the menu loads in before continuing tests?

WunderBart replied:
These kinds of waits are most of the time unnecessary because they're followed by a `clickWhenClickable` call (100% time with any menu element). The `clickWhenClickable` helper will wait until the element is clickable, which is the same as waiting for the menu to be loaded and then calling the click helper. It's better to use the click helper only because we're waiting for the very element we want to interact with. In this case, we were even waiting additionally for some other element (`.is-goup-sites`), whose relation to the menu loading state seems to be unclear. Does that make sense?

noahtallen replied:
Makes sense!

WunderBart committed:
Fix & simplify the sidebar visibility helper
- Sidebar is hidden on mobile viewports only, no need to check it on desktops
- Make the open validator on mobiles stable by checking the `.layout` element: it should have `.focus-sidebar` class for the sidebar to be open.

lynnbgriffith reviewed (approved):
👍 Looks good! Thanks for the continued attention to detail cleaning up the tests!
